### ISSUE TITLE
Returned records not available in afterBulkUpdate

### ISSUE THREAD

**Issue by gabegorelick** [sequelize/sequelize#10202]
## What are you doing?

```js
// assume a model named Project

Project.addHook('afterUpdate', (instance, options) => {
  // do something with instance
});
Project.addHook('afterBulkUpdate', options => {
  // need to fetch instances first, even if we used `returning: true` when calling Project.update
  return Project.findAll({where: options.where})
    .then(instances => {
      // do something with instances
    });
});
```

On Postgres, Sequelize allows us to use `returning` to avoid having to call `findAll` after an `update`. However, even if we return the affected rows, we still need to refetch them since they are not passed to the `afterBulkUpdate` hook: https://github.com/sequelize/sequelize/blob/f2cafcc4b4a2b7500ff672ebdf527f338ee20bb4/lib/model.js#L3008

## What do you expect to happen?

```js
Project.addHook('afterBulkUpdate', (options, affectedRows) => {
   // do something with affectedRows
});
```

Alternatively, an acceptable solution could be to set `individualHooks` when calling `update`. However, this will currently not avoid an extra SELECT because, "this will execute a SELECT followed by individual UPDATEs. A select is needed, because the row data needs to be passed to the hooks." But a select should not be needed for after hooks if we use `returning` since we already got the data!

## What is actually happening?
Need an extra fetch in `afterBulkUpdate` since `affectedRows` is not passed.

__Dialect:__ postgres
__Dialect version:__ any
__Database version:__ any
__Sequelize version:__ master
__Tested with latest release:__ Yes

**Comment by thecpdubguy** [comment]
+1
I am looking for a similar feature, only for `afterBulkDestroy`. Has there been any progress on this?

**Comment by thecpdubguy** [comment]
A workaround I have been able to use is modifying the model options to do the following:
```javascript
        hooks: {
          beforeBulkDestroy: (options) => {
            options.individualHooks = true
          },
          afterDestroy: ({ dataValues }, options) => {
            console.log(dataValues)
          }
        }
```

**Comment by gabegorelick** [comment]
Still an issue

**Comment by thecpdubguy** [comment]
Is there any update on this? I could go poking around to see what I can come up with but I don't know the codebase all that well.

**Comment by papb** [comment]
@thecpdubguy If you could investigate and submit a PR that would be great!

**Comment by JamesJansson** [comment]
My suggestion would be to convert this line: 
https://github.com/sequelize/sequelize/blob/d776d7a54478b2bab7d35db8ff99aafad63cbd9e/lib/model.js#L3241

to
`return this.runHooks('afterBulkUpdate', instances, options).then(() => {`

And update the hooks function accordingly. However, this will definitely cause a breaking change. If I were to do such a pull request, what is the likelihood that it would be accepted?

**Comment by papb** [comment]
@JamesJansson would it work only for postgres? Other dialects would have `instances` set to undefined?

**Comment by JamesJansson** [comment]
@papb You're right. Fundamentally, you could check the mode an behave accordingly. We're probably at the edge of what would be considered functionality useful for a cross-platform ORM. What is the general opinion of how Sequelize treats platform-specific functionality?

**Comment by JamesJansson** [comment]
Maybe to allow backwards compatibility we could use the following format
`return this.runHooks('afterBulkUpdate', options, instances).then(() => {`
at
https://github.com/sequelize/sequelize/blob/d776d7a54478b2bab7d35db8ff99aafad63cbd9e/lib/model.js#L3241

**Comment by papb** [comment]
@JamesJansson I think feature is acceptable, even if it works only for postgres, provided that it does not break compatibility. So I would agree to providing `instances` as a new last argument in the hooks (which shall be undefined for other dialects). What do you think?

**Comment by proxent** [comment]
> ```js
>             options.individualHooks = true
> ```

You saved my day!

**Comment by mistercorea** [comment]
ahh I found a great work around for this.
You try use the `individualHooks: true` to get the individual changes to be fired for afterUpdate.
